034-default-to-restricted-pod-security-profile.md

```diff
@@ -0,0 +1,157 @@
+On OpenShift, the security context is automatically injected into the Pods when they are created based on the OpenShift SCC policies.
+This typically involves dropping some capabilities, using some random unprivileged user ID etc.
+In some cases, our users might also use similar systems to inject the security context using admission controllers and similar tools.
+
+The _baseline_ profile basically means that you just use the default settings without requesting any additional privileges or capabilities.
+So without any of the additional tooling mentioned above, Strimzi operators and the pods created by them today match the _baseline_ profile.
+But if you would try to run Strimzi in a cluster where the _restricted_ profile is enforced, it would not work out of the box.
+
+Apart from the Kaniko builder used for the Kafka Connect build, all our components are able to run under the _restricted_ profile.
+But the operator doesn't configure the appropriate security context and the admission plugin will reject them.
+Users have to _manually_ add the matching security context configuration into the `.template` sections of the custom resources and into the operator deployments to allow the pods to be created under the _restricted_ profile.
+To match the _restricted_ profile, users would need to add the following security context:
+
+```yaml
+          securityContext:
+            allowPrivilegeEscalation: false
+            capabilities:
+              # The restricted allows the use of the NET_BIND_SERVICE capability.
+              # But Strimzi does not require it for anything, so we do not include it here.
+              drop:
+                - ALL
+            runAsNonRoot: true
+            seccompProfile:
+              type: RuntimeDefault
+```
+
+While users can configure the security context, it is not user-friendly.
+The Pod security context does not cover all the different options required by the _restricted_ profile.
+So the security context needs to be basically configured for every container (including init containers, sidecars etc.).
+So just for a single Kafka cluster, it might need to be configured on 9 different places of the Kafka CR (ZooKeeper, Kafka broker, Kafka init-container, Topic Operator, User Operator, TLS sidecar, Kafka Exporter, Cruise Control, another TLS sidecar).
+
+## Proposal
+
+This proposal suggests to make Strimzi more secure by default and configure the security context to match the _restricted_ profile by default.
+This requires changes to the Strimzi Cluster Operator which will set the security context.
+No changes should be required to the operands.
+
+The security context will be set only when running on Kubernetes clusters.
+When OpenShift is detected, the security context will not be set and OpenShift will be able to inject its own security context (unfortunately, the _restricted_ Kubernetes profile is currently not fully compatible with the OpenShift restricted SCC profile).
```

Comment:
How might the approach change should the _restricted_ profile later be compatible with OpenShift? Should that be mentioned here?

Reply:
It would be easier to set it by default even on OpenShift. But I have currently no information that it will change.